src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -0,0 +1,43 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="GetDailyBuildNumber" 
+    AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"
+    Condition="Exists('$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll')" />
+
+  <Target Name="GetDailyBuildNumber"
+    BeforeTargets="Build"
+    Condition="'$(DailyBuildNumGenEnabled)' == 'true'"
+    >
+    <GetDailyBuildNumber
+      VersionPropsFile="$(SourceDir)version.props">
+      <Output PropertyName="DailyBuildNumber" TaskParameter="DailyBuildNumber" />
+    </GetDailyBuildNumber>
+  </Target>
+
+  <Target Name="CommitDailyBuildNumber"
+    AfterTargets="BuildPackages"
+    Condition="'$(DailyBuildNumGenEnabled)' == 'true'"
+    >
```

Comment:
Do you expect DailyBuildNumGenEnabled to only be set via the official build? Perhaps we can call it something better like UpdateDailyBuildNumber or something that indicates that the number will be incremented.